uint16 sp_seq
float64 x
float64 y
float64 z
uint8 action
float64 wait_time
---
bool success
int32 state